clang/test/SemaSYCL/native_cpu/kernelhandler-scalar.cpp

```diff
@@ -0,0 +1,98 @@
+// RUN: %clangxx -fsycl-device-only  -fsycl-targets=native_cpu -Xclang -fsycl-int-header=%t.h -S -o %t.ll %s 
+// RUN: FileCheck -input-file=%t.h.hc %s 
+// RUN: FileCheck -input-file=%t.ll %s --check-prefix=CHECK-LL
+// Compiling generated main integration header to check correctness, -fsycl option used to find required includes
+// RUN: %clangxx -fsycl -D __SYCL_NATIVE_CPU__ -c -x c++ %t.h
```

Comment:
What do you mean by 'check correctness' ? Are you checking that it compiles without error?

Reply:
Yes, this is just to check the integration header is correct for this test case since none of the other invocations compile the header.